jcommune-service/src/main/java/org/jtalks/jcommune/service/bb2htmlprocessors/UrlToLinkConvertPostProcessor.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (C) 2011  JTalks.org Team
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.jtalks.jcommune.service.bb2htmlprocessors;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This text postprocessor search URLs in text and wrap them in &lt;a&gt tag to highlight.
+ * Links which are already inside &lt;a&gt or &lt;img&gt tag are skipped.
+ *
+ * Created by Alexey Usharovskiy on 25.12.16.
+ */
+public class UrlToLinkConvertPostProcessor implements TextPostProcessor {
+
+    private static final Pattern htmlTagsToSkip = Pattern.compile("(<a .*?href=(\"|').*?(\"|')|<img .*?src=(\"|').*?(\"|'))", Pattern.CASE_INSENSITIVE);
+    private static final Pattern urlInText = Pattern.compile("(\\b((?:https?|ftp|file):\\/\\/|www\\.|ftp\\.)[-A-Z0-9+&@#\\/%?=~_|!:,.;]*[-A-Z0-9+&@#\\/%=~_|])", Pattern.CASE_INSENSITIVE);
+    private static final String htmlLinkTemplate = "<a href=\"%s\">%s</a>";
+
+    @Override
+    public String postProcess(String bbDecodedText) {
+        return processLinks(bbDecodedText);
+    }
+
+    /**
+     * Search for text blocks outside &lt;a&gt and &lt;img&gt tags,
+     * pass them to URL process method and build resulting post text
+     * with highlighted URLs
+     *
+     * @param bbEncodedText bb encoded text to process
+     * @return processed text
+     */
+    private static String processLinks(String bbEncodedText) {
+        StringBuilder stringBuilder = new StringBuilder();
+        int prevPos = 0;
+        Matcher matcher = htmlTagsToSkip.matcher(bbEncodedText);
```

Comment:
did you check this postprocessor on the message type of Review (with link into source code) and message type of Discussion  (with link into source code)?

ex. http://qa.jtalks.org/jcommune/topics/10768
      http://qa.jtalks.org/jcommune/topics/10769